doc/amazon_sagemaker_debugger.rst

```diff
@@ -0,0 +1,356 @@
+
+If there are properties you want to apply across all collections, those can be supplied in ``hook_parameters`` within the ``DebuggerHookConfig`` object. For example, to apply a value of ``save_interval`` across all collections do:
+
+.. code:: python
+
+    from sagemaker.debugger import CollectionConfig, DebuggerHookConfig
+
+    collection_config_1 = CollectionConfig(
+        name='collection_name_1',
+        parameters={
+            "include_regex": ".*"
+        }
+    )
+    collection_config_2 = CollectionConfig(
+        name='collection_name_2',
+        parameters={
+            "include_regex": ".*"
+        }
+    }
+
+    debugger_hook_config = DebuggerHookConfig(
+        s3_output_path='s3://path/for/data/emission',
+        container_local_output_path='/local/path/for/data/emission'
+        hook_parameters={
+            "save_interval": "10"
+        },
+        collection_configs=[
+            collection_config_1, collection_config_2
+        ]
+    )
+
+In the above sample code, the ``save_interval`` of ``10`` will be applied for storing both collections.
+
+Note that the value set in the ``collection_parameters`` for a parameter will override the corresponding value of the same parameter in the ``hook_parameters``. For example, in the above sample code, if ``collection_config_2`` had the value of ``save_interval`` set something other than ``10``, say, ``20``, then the tensors for that collection would have been saved with step interval ``20`` while those for ``collection_config_1`` would still be saved with ``10``.
+
+Continuous analyses through Rules
+=================================
+
+In addition to collecting the debugging data, Amazon SageMaker Debugger provides the capability for you to analyze it in a streaming fashion through "rules". SageMaker Debugger 'Rules' are pieces of code which encapsulate the logic for analyzing the debugging data.
+
+SageMaker Debugger provides a set of built-in rules curated by data scientists and engineers at Amazon. There is additional support for bringing in custom rule source codes for evaluation. In the following sections, you'll learn how to use either while training your model.
+
+Relationship between debugger hook and rules
+--------------------------------------------
+
+Using SageMaker Debugger is, broadly, a two-pronged approach. One one hand you have production of debugging data, which is done through the Debugger Hook and on the other hand you have the consumers which can be rules (for continuous analyses) or SageMaker Debugger SDK (for interactive analyses).
+
+Usually, production of data is independent of the manner in which it's consumed (and vice-versa.) For example, you can configure the debugging hook to store only the collection "gradients" and configure the rules to operate on some other collection, say, "weights". While this is possible, it's not ideal since you won't get any meaningful insight into the training process through this.
```

Comment:
move the period after the parenthesis, i.e. s/vice-versa.)/vice-versa).

also, what is the value of including an example that's not actually recommended?

Reply:
Moved the period after the parenthesis.

Which example are you alluding to?

Reply:
I was referring to the suggestion to configure the debugging hook to store "gradients" while the rules operate on "weights" - it read to me a bit like "oh, here's this thing you could do...but actually don't do it because you won't get anything out of it"

Reply:
It's not a suggestion - it's an example of a bad usage. It's more of an FYI because users will inevitably wonder about the relationship of the hook and the rules and it's important to guide them in the right direction by not only listing out correct usages, but equally the wrong/undesired usages.